bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/actions/BaseNotificationActionHandler.java

```diff
@@ -53,5 +64,20 @@ public BaseNotificationActionHandler(Action module, CloudService cloudService) {
 
         Object severityParam = module.getConfiguration().get(PARAM_SEVERITY);
         this.severity = severityParam instanceof String ? severityParam.toString() : null;
+
+        Object onClickActionParam = module.getConfiguration().get(PARAM_ON_CLICK_ACTION);
+        this.onClickAction = onClickActionParam instanceof String ? onClickActionParam.toString() : null;
```

Comment:
Probably https://github.com/openhab/openhab-core/blob/cb65e414453fa6d67634e0db4247bbe9d6071fac/bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/ConfigParser.java could be used here?

This would simplify the code to a few one liners:

```suggestion
        this.onClickAction = ConfigParser.valueOrElse(module.getConfiguration().get(PARAM_ON_CLICK_ACTION), String.class, null);
```

Reply:
Nice, i'll look at cleaning that up, thanks for the pointer.